server/services/SsotToRobotParsing/SsotToRobotParser.js

```diff
@@ -43,24 +72,29 @@ const successorTasksExist = (currentElement) =>
 const writeCodeForElement = (
   id,
   elements,
+  parameters,
+  attributes,
   codeToAppend,
   previousApplication
 ) => {
   const currentElement = elements.find((element) => element.id === id);
-  let combinedCode = codeToAppend
-  let newPreviousApplication = previousApplication
+  let combinedCode = codeToAppend;
+  let newCodeLine = '';
+  let newPreviousApplication = previousApplication;
   if (isAnRpaInstruction(currentElement)) {
-    if (currentElement.rpaApplication !== previousApplication) {
-      combinedCode += `${currentElement.name}\n`;
+    const matchingAttributeObject = attributes.find((attribute) => attribute.activityId === id);
+    if (matchingAttributeObject.rpaApplication !== previousApplication) {
```

Comment:
perhaps we rather call matchingAttributeObject currentAttributeObject

Reply:
same goes for the following lines and there I would also call matchingParameterObject currentParameterObject